monkey/monkey_island/cc/models/monkey.py

```diff
@@ -84,13 +88,52 @@ def get_os(self):
             os = "windows"
         return os
 
+    @staticmethod
+    @ring.lru()
+    def get_label_by_id(object_id):
+        current_monkey = Monkey.get_single_monkey_by_id(object_id)
+        label = Monkey.get_hostname_by_id(object_id) + " : " + current_monkey.ip_addresses[0]
+        if len(set(current_monkey.ip_addresses).intersection(local_ip_addresses())) > 0:
+            label = "MonkeyIsland - " + label
+        return label
+
+    @staticmethod
+    @ring.lru()
+    def get_hostname_by_id(object_id):
+        """
+        :param object_id: the object ID of a Monkey in the database.
+        :return: The hostname of that machine.
+        :note: Use this and not monkey.hostname for performance - this is lru-cached.
+        """
+        return Monkey.get_single_monkey_by_id(object_id).hostname
+
+    def set_hostname(self, hostname):
+        """
+        Sets a new hostname for a machine and clears the cache for getting it.
+        :param hostname: The new hostname for the machine.
+        """
+        self.hostname = hostname
+        self.save()
+        Monkey.get_hostname_by_id.delete(self.id)
+
     def get_network_info(self):
         """
         Formats network info from monkey's model
         :return: dictionary with an array of IP's and a hostname
         """
         return {'ips': self.ip_addresses, 'hostname': self.hostname}
 
+    @staticmethod
+    @ring.lru(
+        expire=1  # data has TTL of 1 second. This is useful for rapid calls for report generation.
+    )
+    def is_monkey(object_id):
```

Comment:
If get_single_monkey_by_id is cached, do we really gain anything by having a cache here?

Reply:
`get_single_monkey_by_id` is not cached (and I'd prefer to keep it this way - more things use it, which will require more testing).